packages/sewing-kit-koa/src/middleware.ts

```diff
@@ -13,6 +13,7 @@ export {Assets, Asset};
 
 export interface Options {
   assetPrefix?: string;
+  assetPrefixCallback?: Function;
```

Comment:
instead of this, I think we'd usually prefer to augment the existing API. I think this would work OK if you made `assetPrefix` be `assetPrefix?: string | (ctx: Context) => string;`

Reply:
This helps avoid the awkward "but what does this mean" situation where both `assetPrefix` and `assetPrefixCallback` are specified.

Reply:
I thought about having the param accept either a string or function but thought it would be weird for a param to be either of two different types. With this approach though, doesn't it mean we would have to do some sort of `typeof` check in order to know how to handle the param?

Reply:
Yeah, you’d use typeof to pick the right type at runtime. In general we opt for this because we try to keep our APIs fairly clean, which generally means one option for a thing (but different choices for what to pass when the API needs more customizability) rather than combinations of multiple options.